larrecodnn/ImagePatternAlgs/Tensorflow/Modules/WaveformRoiFinder_module.cc

```diff
@@ -79,9 +82,18 @@ nnet::WaveformRoiFinder::WaveformRoiFinder(fhicl::ParameterSet const& p)
   fWaveformSize(p.get<unsigned int>("WaveformSize", 6000)),
   fWindowSize(p.get<unsigned int>("ScanWindowSize", 200)),
   fStrideLength(p.get<unsigned int>("StrideLength", 150)),
-  fWireProducerLabel(p.get<art::InputTag>("WireProducerLabel"))
+  fRawProducerLabel(p.get<art::InputTag>("RawProducerLabel","")),
+  fWireProducerLabel(p.get<art::InputTag>("WireProducerLabel",""))
 {
 
+  if (fRawProducerLabel.empty() && fWireProducerLabel.empty()){
+    throw cet::exception("WaveformRoiFinder") << "Both RawProducerLabel and WireProducerLabel are empty";
+  }
+
+  if ((!fRawProducerLabel.empty()) && (!fWireProducerLabel.empty())){
+    throw cet::exception("WaveformRoiFinder") << "Only one of RawProducerLabel and WireProducerLabel should be set";
+  }
```

Comment:
I think there is probably a better way to handle this type of situation--where you only want one or the other, but that would require some more thought.  For now, it's okay.

Reply:
Ok.